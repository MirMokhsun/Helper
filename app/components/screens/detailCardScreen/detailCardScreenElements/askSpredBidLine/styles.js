import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        flex: 1,
        justifyContent: 'space-between'
    },
    textWrapper: {
        flex: 2,
        justifyContent: 'flex-end',
        paddingRight: 10,
        flexDirection: 'row',
    },
    rendersNumbWrapper: {
        flexDirection: 'row',
        justifyContent: 'flex-start',
        flex: 3
    },
    textTitle: {
        fontFamily: 'Roboto-Light',
        lineHeight: 20,
        fontSize: 14,
        color: '#000756',
    },
    spredContainer: {
        borderLeftWidth: 1,
        borderRightWidth: 1,
        borderColor: '#D3D9E7',
        flex: 2,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'space-around'

    },
    spred:{
        fontFamily: 'Roboto-Light',
        lineHeight: 25,
        fontSize: 14,
        color: '#9FA0B2',
    },
    numberContainer: {
        flex: 3,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'space-around'

    },
})