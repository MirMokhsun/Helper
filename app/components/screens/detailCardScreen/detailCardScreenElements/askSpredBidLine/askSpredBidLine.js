import React from 'react';
import { View, Text } from 'react-native';
import { styles } from './styles';
import { QuotationItemDetailCard } from '../quotationItemDetailCard/quotationItemDetailCard';
import { getSpred } from '../../../../../api/openCardLogic/openCardApi';

export class AskSpredBidLine extends React.PureComponent {
    constructor(props) {
        super(props)

        this.state = {
            spred: 0
        }
    }

    componentWillMount() {
        const { bid, ask } = this.props;
        const spred = getSpred(ask, bid);
        this.setState({ spred });
    }

    render() {
        const { bid, ask } = this.props;
        const { spred } = this.state;
        return (
            <View style={styles.container}>
                <View style={styles.numberContainer}>
                    <Text style={styles.textTitle}>Ask</Text>
                    <QuotationItemDetailCard number={(ask == 0) ? '--' : ask} />
                </View>
                <View style={styles.spredContainer}>
                    <Text style={styles.textTitle}>Spred</Text>
                    <Text style={styles.spred}>{spred}</Text>
                </View>
                <View style={styles.numberContainer}>
                    <Text style={styles.textTitle}>Bid</Text>
                    <QuotationItemDetailCard number={(bid == 0) ? '--' : bid} />
                </View>
            </View >
        )
    }
}

export default AskSpredBidLine;