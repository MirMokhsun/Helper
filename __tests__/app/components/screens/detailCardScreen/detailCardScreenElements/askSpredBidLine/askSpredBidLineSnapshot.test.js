import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AskSpredBidLine } from '../../../../../../../app/components/screens/detailCardScreen/detailCardScreenElements/askSpredBidLine/askSpredBidLine';

it('create snapshot AskSpredBidLine', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<AskSpredBidLine ask={0.9454} bid={0.9545} language={'ru'} />
    );
    expect(toolbarSnapshot).toMatchSnapshot();
});