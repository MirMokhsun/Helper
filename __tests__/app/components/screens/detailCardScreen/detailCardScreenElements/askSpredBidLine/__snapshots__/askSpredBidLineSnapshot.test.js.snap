// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot AskSpredBidLine 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000756",
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
    >
      Ask
    </Text>
    <QuotationItemDetailCard
      number={0.9454}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#D3D9E7",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "flex": 2,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000756",
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
    >
      Spred
    </Text>
    <Text
      style={
        Object {
          "color": "#9FA0B2",
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "lineHeight": 25,
        }
      }
    >
      -9.1
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000756",
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
    >
      Bid
    </Text>
    <QuotationItemDetailCard
      number={0.9545}
    />
  </View>
</View>
`;
